web/packages/teleport/src/UnifiedResources/ResourceCard.tsx

```diff
@@ -132,17 +140,44 @@ export function ResourceCard({ resource, onLabelClick }: Props) {
     };
   });
 
+  // Clear the timeout on unmount to prevent changing a state of an unmounted
+  // component.
+  useEffect(() => () => clearTimeout(collapseTimeout.current), []);
+
   const onMoreLabelsClick = () => {
     setShowAllLabels(true);
   };
 
+  const onMouseLeave = () => {
+    // If the user expanded the labels and then scrolled down enough to hide the
+    // top of the card, we scroll back up and collapse the labels with a small
+    // delay to keep the user from losing focus on the card that they were
+    // looking at. The delay is picked by hand, since there's no (easy) way to
+    // know when the animation ends.
```

Comment:
I came to the PR here looking for "why is there a delay for this card closing?" because it was jarring to me, but after reading this comment, I agree with the approach. Thanks for changing my mind